#!/usr/bin/env python

PACKAGE = "wave_mission"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("mission_name",    str_t,    0, "Name of mission", "NULL")
gen.add("mission_enabled", bool_t, 0, "The execution state of the mission", False)
gen.add("path_tolerance",    double_t,    0, "Path tolerance",  1.0, 0.1, 10.0)
gen.add("waypoint_tolerance",   double_t,   0, "Waypoint tolerance",  1.0, 0.1, 10.0)
gen.add("cruising_speed",   double_t,   0, "Cruising speed",  0.5, 0.01, 3.0)
gen.add("max_speed",   double_t,   0, "Max speed",  1.0, 0.01, 3.0)

exit(gen.generate(PACKAGE, "wave_mission", "WaveMission"))
